Lib/asyncio/base_events.py

```diff
@@ -155,6 +156,63 @@ def _run_until_complete_cb(fut):
     futures._get_loop(fut).stop()
 
 
+
+class _SendfileProtocol(protocols.Protocol):
+    def __init__(self, transp):
+        # transport should be _FlowControlMixin instance
```

Comment:
Can you add an `if not isinstance(transp, _FlowControlMixin): raise TypeError`. I know that it shouldn't be possible to get this exception, but in case we made a mistake and some user hits this error message, it would be more informative than some random `AttributeError`.

Reply:
Perform the check in debug mode only maybe?

Reply:
`sendfile` isn't a frequent operation so one `isinstance` call won't slow it down, so let's check it always.